### PR TITLE
[pcre2] Fix "There should be no absolute paths"

### PR DESCRIPTION
**Pull request by bansan85**
Problem occurs also on x64-linux-dynamic.

```
warning: There should be no absolute paths, such as the following, in an installed package:
  /home/XXX/prog/vcpkg/packages/pcre2_x64-linux-dynamic
  /home/XXX/prog/vcpkg/installed
  /home/XXX/prog/vcpkg/buildtrees/pcre2
  /home/XXX/prog/vcpkg/downloads
Absolute paths were found in the following files:
  /home/XXX/prog/vcpkg/packages/pcre2_x64-linux-dynamic/bin/pcre2-config
  /home/XXX/prog/vcpkg/packages/pcre2_x64-linux-dynamic/debug/bin/pcre2-config
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: /home/XXX/prog/vcpkg/ports/pcre2/portfile.cmake
```

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- ~~[ ] The "supports" clause reflects platforms that may be fixed by this new version.~~
- ~~[ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- ~~[ ] Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/pcre2/portfile.cmake [hunk 1]
```diff
@@ -62,7 +62,7 @@ file(REMOVE_RECURSE
 
 if(BUILD_STATIC)
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-elseif(VCPKG_TARGET_IS_WINDOWS)
+else()
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/..")
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/pcre2/portfile.cmake:66]: Yes, and for both types of linkage.

[the remark this answers is not in the document]
Comment (bansan85) [reply ports/pcre2/portfile.cmake:66]: @Osyotr @dg0yt Are you sure ? `xml2-config` is also in `bin`.

And there is no folder `tools` for `debug`.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/pcre2/portfile.cmake:66]: Move or remove. A typical drive-by fix for inconsistencies. Windows bias, I guess, plus lack of specification, plus mostly replaced by pkg-config files (or cmake config).

You can use `tools/pcre2/debug` for the debug variant. libxslt does that. xapian and curl with a small variant.

**Comment by bansan85** [comment]
Move pcre2-config to tools/pcre2 and tools/pcre2/debug based on libxslt port.

File: ports/pcre2/portfile.cmake [hunk 2]
```diff
@@ -60,14 +60,16 @@ file(REMOVE_RECURSE
...
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/pcre2")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}/tools/pcre2/pcre2-config")
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/pcre2/pcre2-config" "${CURRENT_PACKAGES_DIR}" [[$(cd "$(dirname "$0")/../.."; pwd -P)]])
+if(NOT VCPKG_BUILD_TYPE)
+    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/pcre2/debug")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}/tools/pcre2/debug/pcre2-config")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/pcre2/debug/pcre2-config" "${CURRENT_PACKAGES_DIR}/debug" [[$(cd "$(dirname "$0")/../../../debug"; pwd -P)]])
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/pcre2/debug/pcre2-config" [[${prefix}/include]] [[${prefix}/../include]])
 endif()
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin" "${CURRENT_PACKAGES_DIR}/bin")
```
Comment (Osyotr) [inline ports/pcre2/portfile.cmake:72]: What about windows?

Comment (bansan85) [reply ports/pcre2/portfile.cmake]: Sigh... Sorry, I tried only on Linux (static and debug) and didn't think for Windows shared that stores library in bin folder.

I added a condition about static linkage like other ports does.

**Comment by bansan85** [comment]
I don't know what's wrong. I run with success `vcpkg install colmap` for triplet x64-linux without error.

`opencv4[ade,contrib,core,dc1394,default-features,dnn,eigen,ffmpeg,freetype,gdcm,gtk,ipp,jasper,jpeg,lapack,nonfree,openexr,opengl,openmp,png,python,qt,quirc,sfm,tiff,vtk,webp]` is in progress.

**Comment by dg0yt** [comment]
Baseline regression around lapack: https://github.com/microsoft/vcpkg/pull/38100

**Comment by bansan85** [comment]
> That is a different error than the one I saw. That looks like something is missing to declare GKlib as a transitive usage requirement

Ok. I tried to found where the problem is.  I have some hint but didn't found the good patch.

**Comment by dg0yt** [comment]
> That is a different error than the one I saw. That looks like something is missing to declare GKlib as a transitive usage requirement

Or maybe still related via g2o? https://github.com/microsoft/vcpkg/issues/33041

**Comment by Neumann-A** [comment]
> Or maybe still related via g2o? https://github.com/microsoft/vcpkg/issues/33041

That just means g2o has more than one problem ;) I only fixed the blas issue I observed

**Comment by bansan85** [comment]
The problem with `rtabmap` came from `ports/metis/build-fixes.patch`. `metis` now depends on `GKlib`. So now, `rtabmap_core` needs to depend on `PUBLIC metis` and not `PRIVATE metis`.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**